components/organisms/messaging/components/voiceRecorder/index.js

```diff
@@ -0,0 +1,110 @@
+// React
+import { useEffect, useRef, useState } from 'react'
+import { MessageNames, MessagingCommunicationService, MessagingActions } from 'components/services'
+import styled, { css } from 'styled-components'
+import { Microphone } from './microphone'
+import { Close, Text } from 'components'
+
+let AudioRecorder
+let mpegEncoder
+
+export const VoiceRecorder = props => {
+  const [recorderLoaded, setRecorderLoaded] = useState(false)
+
+  const recorder = useRef()
+
+  const canSendData = useRef(true)
+
+  const [isRecording, setisRecording] = useState(false)
+
+  const loadModules = async () => {
+    AudioRecorder = (await import('audio-recorder-polyfill')).default
+    mpegEncoder = (await import('audio-recorder-polyfill/mpeg-encoder')).default
+    AudioRecorder.encoder = mpegEncoder
+    AudioRecorder.prototype.mimeType = 'audio/mpeg'
+    if (window) {
+      window.MediaRecorder = AudioRecorder
+    }
+    setRecorderLoaded(true)
+  }
+
+  useEffect(() => {
+    loadModules()
+  }, [])
+
+  const handleStartRecord = () => {
+    navigator.mediaDevices.getUserMedia({ audio: true }).then(stream => {
```

Comment:
I guess you're using `window.navigator` here, so isn't it better to make sure if we always have access to `navigator`? (from `SSR` aspect)

Reply:
because this component is loaded lazily, this is not necessary, it will only be mounted on the client machine, but it is good to use ? anyway, thanks for noticing